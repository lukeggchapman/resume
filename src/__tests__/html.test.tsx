import React from 'react'
import { shallow } from 'enzyme'

import HTML, { HTMLProps } from '../html'

const baseProps: HTMLProps = {
  body: 'bodyTest',
  bodyAttributes: {
    bodyAttributes: 'test',
  },
  headComponents: 'headComponentsTest',
  htmlAttributes: {
    htmlAttributes: 'test',
  },
  postBodyComponents: 'postBodyComponentsTest',
  preBodyComponents: 'preBodyComponentsTest',
}

describe('HTML', () => {
  const render = () => shallow(<HTML {...baseProps} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
