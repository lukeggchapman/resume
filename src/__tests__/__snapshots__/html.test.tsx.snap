// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML renders correctly 1`] = `
<html
  css={
    Array [
      "
        height: 100%;
        overflow: hidden;
      ",
    ]
  }
  htmlAttributes="test"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="ie=edge"
      httpEquiv="x-ua-compatible"
    />
    <meta
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
      name="viewport"
    />
    headComponentsTest
  </head>
  <body
    bodyAttributes="test"
    css={
      Array [
        "
          height: 100%;
        ",
      ]
    }
  >
    preBodyComponentsTest
    <noscript
      id="gatsby-noscript"
      key="noscript"
    >
      This app works best with JavaScript enabled.
    </noscript>
    <div
      css={
        Array [
          "
            height: 100%;
            > div[role='group'][tabindex] {
              height: 100%;
            }
          ",
        ]
      }
      dangerouslySetInnerHTML={
        Object {
          "__html": "bodyTest",
        }
      }
      id="___gatsby"
      key="body"
    />
    postBodyComponentsTest
  </body>
</html>
`;
